### PR TITLE
chore: use mocks for providers

### LINKED ISSUE #791: Use mock for catalog providers in unit tests
Instead of using test/dummy provider, generate mocks and use in tests.

### PR DESCRIPTION
**Pull request by rollandf**
Fix #791

### CODE REVIEW

File: .mockery.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+inpackage: False
+testonly: False
+with-expecter: True
+keeptree: True
```
Comment (adrianchiris) [inline .mockery.yaml:4]: nit: newline

**Review by adrianchiris — DISMISSED** [review]

File: .mockery.yaml (hunk 1 above)
Comment (rollandf) [reply .mockery.yaml]: done

**Review by vasrem — APPROVED** [review]

**Comment by rollandf** [comment]
Copy Right Validation is checking also auto-generated files...
Maybe we should update this job to filter those somehow.

**Review by e0ne — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**